core/server/src/main/java/alluxio/web/UIWebServer.java

```diff
@@ -65,29 +69,36 @@ public UIWebServer(ServiceType service, InetSocketAddress address) {
     QueuedThreadPool threadPool = new QueuedThreadPool();
     int webThreadCount = Configuration.getInt(PropertyKey.WEB_THREADS);
 
-    mServer = new Server();
-    SelectChannelConnector connector = new SelectChannelConnector();
-    connector.setHost(address.getHostName());
-    connector.setPort(address.getPort());
-    connector.setAcceptors(webThreadCount);
-    mServer.setConnectors(new Connector[] {connector});
-
-    try {
-      mServer.getConnectors()[0].close();
-      mServer.getConnectors()[0].open();
-    } catch (IOException e) {
-      throw Throwables.propagate(e);
-    }
-
     // Jetty needs at least (1 + selectors + acceptors) threads.
     threadPool.setMinThreads(webThreadCount * 2 + 1);
     threadPool.setMaxThreads(webThreadCount * 2 + 100);
-    mServer.setThreadPool(threadPool);
+
+    mServer = new Server(threadPool);
+
+    mServerConnector = new ServerConnector(mServer);
+    mServerConnector.setPort(mAddress.getPort());
+    mServerConnector.setHost(mAddress.getAddress().getHostAddress());
+
+    mServer.addConnector(mServerConnector);
```

Comment:
It seems kind of weird since opening the connector without the server running seems incorrect? Though technically our RPC server has the same problem due to how it is implemented. I'm open to changing it if you think the consistency is valuable. Ultimately I think there should be a utils method which can get an ephemeral port so we don't need to rely on the implementation of the servers we want to use to automatically get one.